image/tarexport/load.go

```diff
@@ -422,8 +422,6 @@ func checkCompatibleOS(imageOS string) error {
 		return fmt.Errorf("cannot load %s image on %s", imageOS, runtime.GOOS)
 	}
 	// Finally, check the image OS is supported for the platform.
-	if err := system.ValidatePlatform(system.ParsePlatform(imageOS)); err != nil {
-		return fmt.Errorf("cannot load %s image on %s: %s", imageOS, runtime.GOOS, err)
-	}
+	// TODO(@arm64b): Leave this sanity check to the containerd code in the future
```

Comment:
Looks like these comments are no longer current?

Reply:
IMO, except the `parser` to parse the `os/arch/variant/[os version, os feature, ...], seems we need a sanity checker for the parsed unit to offload the `parser` in containerd code, that's why I have a `TODO` here. But I am not sure if it's 100% necessary.

Reply:
In my PR #37346 , I have below workaround 😄:
```
+// TODO(@arm64b): place holder for the sanity check of the spec platform
 +func ValidatePlatform(platform *specs.Platform) error {
 +	return nil
 +}
```
This code should be in the containerd as the partner of `Parse` function.

Reply:
Currently containerd doesn't have this function. I'm not sure how they would implement it and it would need a PR there before. Some of what the old  function did is handled automatically in `Parse`. I think it would make sense to validate lcow in windows like it was before so I could add it back. Ideally, we would add supported platforms into daemon config so we can validate the linux cases as well (like the emulators) and correctly pick what nodes can handle the workloads. Eg. https://github.com/moby/buildkit/blob/master/solver/llbsolver/vertex.go#L77 is supported in buildkit.

@thaJeztah So in what state do you want me to leave this. Just remove comments for now or something more?

Reply:
I think it makes sense to keep something in there temporarily, but not critical. I'm not sure how misleading the error message would be without the check though.

Reply:
Added back the lcow-only validation.